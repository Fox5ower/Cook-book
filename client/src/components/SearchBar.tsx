import React from "react";
import IDish from "../interfaces/IDish";
import { FormattedMessage } from "react-intl";

interface MyProps {
    term: string,
    data: Array<IDish>,
    filteredDish: Array<IDish>,
    update: Function
}

const Searchbar: React.SFC<MyProps> = ({ term, data, filteredDish, update }) => {

    const dataSearch = (e: any) => {
        const value = e.target.value.toLowerCase();
        if (filteredDish.length >= 1) {
            const filter = filteredDish.filter((dish: IDish) => {
                return dish.name.toLowerCase().includes(value);
            });
            update({
                dishes: filter,
                term: value
            });
        } else {
            const filter = data.filter((dish: IDish) => {
                return dish.name.toLowerCase().includes(value);
            });
            update({
                dishes: filter,
                term: value
            });
        }



    };

    return (
        <div className="search-icon">
            <FormattedMessage id="dishes.search.placeholder" defaultMessage="Search by name...">
                {(placeholder: string) =>
                    <input data-testid="search-bar"
                        value={term}
                        placeholder={placeholder}
                        onChange={dataSearch}
                        type="text" required
                    />
                }
            </FormattedMessage>
        </div>
    );
};

export default Searchbar