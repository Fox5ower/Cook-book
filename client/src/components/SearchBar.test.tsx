import React from "react";
import ReactDOM from "react-dom";
import SearchBar from "./SearchBar";
import renderer from "react-test-renderer";
import { render, cleanup } from "@testing-library/react";
import { IntlProvider } from "react-intl";
import English from "../languages/en.json";

afterEach(cleanup)

const update = (): void => { }


it("Should renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<IntlProvider locale="en" messages={English}><SearchBar
        term="Test"
        data={[]}
        filteredDish={[]}
        update={update}
    /></IntlProvider>, div);
})

it("Should renders searchBar correctly", () => {
    const { getByTestId } = render(<IntlProvider locale="en" messages={English}><SearchBar
        term="Test"
        data={[]}
        filteredDish={[]}
        update={update}
    /></IntlProvider>)

    expect(getByTestId("search-bar")).toHaveValue("Test");
})

it("Should match the Snapshot", () => {
    const tree = renderer.create(<IntlProvider locale="en" messages={English}><SearchBar
        term="Test"
        data={[]}
        filteredDish={[]}
        update={update}
    /></IntlProvider>).toJSON();

    expect(tree).toMatchSnapshot();
})