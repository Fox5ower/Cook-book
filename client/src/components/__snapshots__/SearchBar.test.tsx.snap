// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the Snapshot 1`] = `
<div
  className="search-icon"
>
  <input
    data-testid="search-bar"
    onChange={[Function]}
    placeholder="Search by name..."
    required={true}
    type="text"
    value="Test"
  />
</div>
`;
